src/collective/taxonomy/tests/test_indexer.py

```diff
@@ -0,0 +1,61 @@
+# -*- coding: utf8 -*-
+from collective.taxonomy.testing import INTEGRATION_TESTING
+from collective.taxonomy.interfaces import ITaxonomy
+from plone import api
+from plone.app.testing import applyProfile
+from plone.schemaeditor.utils import FieldAddedEvent
+from plone.schemaeditor.utils import IEditableSchema
+from zope import schema
+from zope.component import queryUtility
+from zope.event import notify
+from zope.lifecycleevent import ObjectAddedEvent
+import unittest
+
+
+class TestIndexer(unittest.TestCase):
+
+    """Test JSON views."""
+
+    layer = INTEGRATION_TESTING
+
+    def setUp(self):
+        self.portal = self.layer['portal']
+        self.portal.portal_workflow.setDefaultChain(
+            'simple_publication_workflow')
+        applyProfile(self.portal, 'plone.app.contenttypes:plone-content')
+        self.document = api.content.create(
+            container=self.portal, type='Document', title='Doc')
+
+    def test_indexer_with_field(self):
+        portal_catalog = api.portal.get_tool('portal_catalog')
+        utility = queryUtility(ITaxonomy, name='collective.taxonomy.test')
+        taxonomy = utility.data
+        taxonomy_test = schema.List(
```

Comment:
I think you actually wanted schema.Set instead of schema.List here. schema.Set is what get when you create a list of choice ttw if I remember and that was your problem.